nxdrive/session_csv.py

```diff
@@ -0,0 +1,68 @@
+import csv
+from pathlib import Path
+from typing import Any, Dict, List
+
+from nxdrive.objects import Session
+
+from .engine.engine import Engine
+from .manager import Manager
+
+
+class SessionCsv:
+    """
+    Class to create a CSV from a Direct Transfer session.
+
+    Usage:
+
+        session_csv = SessionCsv(manager, engine, session)
+        session_csv.generate(session_items)
+        output_path = session_csv.output_file
+    """
+
+    def __init__(self, manager: "Manager", engine: Engine, session: Session, /) -> None:
+        self._manager = manager
+        self._engine = engine
+        self._session = session
+
+        name = f"session_{session.completed_on.replace(':', '-').replace(' ', '_')}"
+        self.output_file = Path(self._manager.home / "csv" / name).with_suffix(".csv")
+        self.output_tmp = Path(self._manager.home / "csv" / name).with_suffix(".tmp")
```

Comment:
Simplify and minor optimization:
```suggestion
        name = f"session_{session.completed_on.replace(':', '-').replace(' ', '_')}.csv"
        self.output_file = self._manager.home / "csv" / name
        self.output_tmp = self.output_file.with_suffix(".tmp")
```